### DISCUSSION TITLE
creating-a-dotnet-asciimath-parser-and-using-it-in-a-markdig-extension/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#418] [Announcements]
# creating-a-dotnet-asciimath-parser-and-using-it-in-a-markdig-extension/

In this post I describe how I ported a Ruby AsciiMath implementation to .NET, created the AsciiMath NuGet package, and used it to create a Markdig extension

https://andrewlock.net/creating-a-dotnet-asciimath-parser-and-using-it-in-a-markdig-extension/

**Comment by wangengzheng** [comment]
great

**Comment by Nicelittlewood** [comment]
Great work.
I've implemented and it works very well, but i've an issue with all command like "\text{something}"
it simply not display anithing. 
Seems non work as well with every "\" command so to achive my result i have to replace all with nothing. 
It's not the most elegant solution, but works.

Waiting for update!

**Reply by Nicelittlewood** [reply]
It would be nice if the "\(...\)" was recognized as well as the "$$...$$" as Math script
